packages/app/src/firebaseServerApp.ts

```diff
@@ -0,0 +1,133 @@
+import { FirebaseAppImpl } from './firebaseApp';
+import { ERROR_FACTORY, AppError } from './errors';
+
+export class FirebaseServerAppImpl
+  extends FirebaseAppImpl
+  implements FirebaseServerApp
+{
+  private readonly _serverConfig: FirebaseServerAppSettings;
+  private _finalizationRegistry: FinalizationRegistry<object>;
+  private _refCount: number;
+
+  constructor(
+    options: FirebaseOptions | FirebaseAppImpl,
+    serverConfig: FirebaseServerAppSettings,
+    name: string,
+    container: ComponentContainer
+  ) {
+    // Build configuration parameters for the FirebaseAppImpl base class.
+    const automaticDataCollectionEnabled =
+      serverConfig.automaticDataCollectionEnabled !== undefined
+        ? serverConfig.automaticDataCollectionEnabled
+        : false;
+
+    // Create the FirebaseAppSettings object for the FirebaseAppImp constructor.
+    const config: Required<FirebaseAppSettings> = {
+      name,
+      automaticDataCollectionEnabled
+    };
+
+    if ((options as FirebaseOptions).apiKey !== undefined) {
+      // Construct the parent FirebaseAppImp object.
+      super(options as FirebaseOptions, config, container);
+    } else {
+      const appImpl: FirebaseAppImpl = options as FirebaseAppImpl;
+      super(appImpl.options, config, container);
+    }
+
+    // Now construct the data for the FirebaseServerAppImpl.
+    this._serverConfig = {
+      automaticDataCollectionEnabled,
+      ...serverConfig
+    };
+
+    this._finalizationRegistry = new FinalizationRegistry(() => {
+      this.automaticCleanup();
+    });
+
+    this._refCount = 0;
+    this.incRefCount(this._serverConfig.releaseOnDeref);
+
+    // Do not retain a hard reference to the dref object, otherwise the FinalizationRegisry
+    // will never trigger.
+    this._serverConfig.releaseOnDeref = undefined;
+    serverConfig.releaseOnDeref = undefined;
+  }
+
+  get refCount(): number {
+    return this._refCount;
+  }
+
+  // Increment the reference count of this server app. If an object is provided, register it
+  // with the finalization registry.
+  incRefCount(obj: object | undefined): void {
+    if (this.isDeleted) {
+      return;
+    }
+    this._refCount++;
+    if (obj !== undefined) {
+      this._finalizationRegistry.register(obj, this);
+    }
+  }
+
+  // Decrement the reference count.
+  decRefCount(): number {
+    if (this.isDeleted) {
+      return 0;
+    }
+    return --this._refCount;
+  }
+
+  // Invoked by the FinalizationRegistry callback to note that this app should go through its
+  // reference counts and delete itself if no reference count remain. The coordinating logic that
+  // handles this is in deleteApp(...).
+  private automaticCleanup(): void {
+    void deleteApp(this);
+  }
+
+  get settings(): FirebaseServerAppSettings {
+    this.checkDestroyed();
+    return this._serverConfig;
+  }
+
+  authIdTokenVerified(): Promise<void> {
+    this.checkDestroyed();
+    // TODO
+    return Promise.resolve();
```

Comment:
Please use Promise.reject() until I wrap up #8008

Reply:
I think I'm going to remove the method altogether.

Reply:
I'll wait to see how much progress you can make today!